pandas/util/testing.py

```diff
@@ -639,10 +761,42 @@ def assert_numpy_array_equal(np_array, assert_equal,
     equivalent if the arrays have equal non-NaN elements,
     and `np.nan` in corresponding locations.
     """
-    if array_equivalent(np_array, assert_equal, strict_nan=strict_nan):
+
+    # compare shape and values
+    if array_equivalent(left, right, strict_nan=strict_nan):
         return
+
     if err_msg is None:
-        err_msg = '{0} is not equivalent to {1}.'.format(np_array, assert_equal)
+        # show detailed error
+
+        if np.isscalar(left) and np.isscalar(right):
+            # show scalar comparison error
+            assert_equal(left, right)
+        elif is_list_like(left) and is_list_like(right):
+            # some test cases pass list
+            left = np.asarray(left)
+            right = np.array(right)
+
+            if left.shape != right.shape:
+                raise_assert_detail(obj, '{0} shapes are different'.format(obj),
+                                    left.shape, right.shape)
+
+            diff = 0
+            for l, r in zip(left, right):
+                # count up differences
+                if not array_equivalent(l, r, strict_nan=strict_nan):
+                    diff += 1
+
+            diff = diff * 100.0 / left.size
+            msg = '{0} values are different ({1} %)'.format(obj, np.round(diff, 5))
+            raise_assert_detail(obj, msg, left, right)
+        elif is_list_like(left):
```

Comment:
Added deatailed error messages and tests to be compat with``assert_almost_equal``.